src/main/java/com/aws/greengrass/util/GreengrassServiceClientFactory.java

```diff
@@ -158,8 +165,9 @@ public GreengrassV2DataClient getGreengrassV2DataClient() {
     /**
      * Initializes and returns GreengrassV2DataClient.
      * @throws DeviceConfigurationException when fails to validate configs.
+     * @throws TLSAuthException when fails to configure the client
      */
-    public GreengrassV2DataClient fetchGreengrassV2DataClient() throws DeviceConfigurationException {
+    public GreengrassV2DataClient fetchGreengrassV2DataClient() throws DeviceConfigurationException, TLSAuthException {
```

Comment:
How would the caller handle TLSAuthException?

Reply:
For any other cause of `TLSAuthException` (other than `ServiceUnavailableException`), we do not retry are the errors are deterministic.  
- In Nucleus,  as we cannot create a GGV2 data client, deployments fail and we will not retry these again. We will not also cache credentials. But nucleus can still receive deployments and try caching credentials like normal. 
- In Shadow manager and CDA, requests due to these exceptions are not retried but are handled.